website/server/controllers/api-v3/members.js

```diff
@@ -57,6 +57,288 @@ api.getMember = {
+}
+
+
+function _addPluralAchievement (res, memberObj, achievementsResult, key, icon, category, keyToCheck, labels) {
+  if (!keyToCheck)
+    keyToCheck = key;
+
+  let value = memberObj.achievements[keyToCheck];
+  let isEarned = !!value;
+
+  // value == 0, labels.singularTitle, labels.singularText
+  // value != 0, labels.pluralTitle, labels.pluralText
+
+  if(!labels){
+    labels = {
+      singularTitle: key+'Singular',
+      singularText: key+'SingularText',
+
+      pluralTitle: key+'Name',
+      pluralText: key+'Text'
+    };
+  }
+
+  let title = res.t(value == 0 ? labels.singularTitle : labels.pluralTitle, {count: value});
+  let text = res.t(value == 0 ? labels.singularText : labels.pluralText, {count: value});
+
+  _addAchievement(achievementsResult, 'plural', key, title, text, icon, category, value, isEarned);
+}
+
+function _addUltimateGearAchievement (res, memberObj, achievementsResult, key, icon, category, keyToCheck) {
+  if (!keyToCheck)
+    keyToCheck = key;
+
+  let value = memberObj.achievements.ultimateGearSets[keyToCheck];
+  let isEarned = !!value;
+
+  let title = res.t('ultimGearName', {ultClass: res.t(key)});
+  let text = res.t(key + 'UltimGearText');
+
+  _addAchievement(achievementsResult, 'ultimateGear', 'ultimateGear' + key, title, text, icon, category, value, isEarned);
+}
+
+/**
+ * @api {get} /api/v3/members/:memberId/achievements Get member achievements object
+ * @apiVersion 3.0.0
+ * @apiName GetMemberAchievements
+ * @apiGroup Member
+ *
+ * @apiParam {UUID} memberId The member's id
+ *
+ * @apiSuccess {Object} data The Achievements object
+ */
+api.getMemberAchievements = {
+  method: 'GET',
+  url: '/members/:memberId/achievements',
+  middlewares: [],
+  async handler (req, res) {
+    req.checkParams('memberId', res.t('memberIdRequired')).notEmpty().isUUID();
+
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let memberId = req.params.memberId;
+
+    let member = await User
+      .findById(memberId)
+      .select(memberFields)
+      .exec();
+
+    if (!member) throw new NotFound(res.t('userWithIDNotFound', {userId: memberId}));
+
+    let achievements = fillAchievements(member, res);
+
+    res.respond(200, achievements);
+  },
+};
+
+// copy from rootCtrl => $scope.contribText
+function contribText(contrib, backer, res){
+  if (!contrib && !backer) return;
+  if (backer && backer.npc) return backer.npc;
+  var l = contrib && contrib.level;
+  if (l && l > 0) {
+    var level = (l < 3) ? res.t('friend') : (l < 5) ? res.t('elite') : (l < 7) ? res.t('champion') : (l < 8) ? res.t('legendary') : (l < 9) ? res.t('guardian') : res.t('heroic');
+    return level + ' ' + contrib.text;
+  }
+}
+
+function fillAchievements (member, res) {
+  var achievements = {
+    _count: 0,
+  };
+
+  // Basic Achievements
+  _addPluralAchievement(res, member, achievements, 'streak', 'achievements-thermometer', 'Basic');
```

Comment:
I think most (if not all) of these sprite names should be `'achievement-{achiev}'`, singular "achievement" instead of plural "achievements".